metadata-db-app/src/cmr/metadata_db/services/concept_service.clj

```diff
@@ -334,7 +516,20 @@
           provider
           concept
           rollback-fn)
-        concept)
+
+        ;; Perform post commit variable association
+        (if (and (= :variable (:concept-type concept))
+                 (:coll-concept-id concept)
```

Comment:
Regardless how you get here, when the variable concept is to be inserted in Oracle, the concept must have the collection concept id, otherwise it would be invalid.

Reply:
OK. got it. You are saying right now there is a need for this check because we have not removed the old variable update endpoint. Once we remove the old variable update endpoint, we would be able to remove this check.

Reply:
Old variable endpoint will not go away, as a mean of updating the variable alone, without updating the association. so this check will stay.